### PR TITLE
Make gc-base, gc-iceberg classes Java 8 compatible

### PR DESCRIPTION
**Pull request by ajantha-bhat**
GC modules can be used by engines to build its own distributed GC logic. Hence, it has to be java8 compatible.

### CODE REVIEW

**Review by snazy — CHANGES_REQUESTED** [review]
TL;DR this is way too much, doing way more than necessary

**Comment by ajantha-bhat** [comment]
> TL;DR this is way too much, doing way more than necessary

Goal was to make gc-base and gc-iceberg compatible with java 8. 
So, I changed `nessie-conventions-iceberg` to use java 8 and then started fixing its dependencies compatibility. 

May be you meant introduce a new profile `nessie-conventions-iceberg8` and use for only `gc-base` and `gc-iceberg` and its dependencies. I will give it a try.

**Comment by ajantha-bhat** [comment]
> TL;DR this is way too much, doing way more than necessary

Updated.

**Review by snazy — APPROVED** [review]
